asg/src/expression/array_inline.rs

```diff
@@ -66,10 +66,12 @@ impl<'a> ExpressionNode<'a> for ArrayInlineExpression<'a> {
     }
 
     fn get_type(&self) -> Option<Type<'a>> {
-        Some(Type::Array(
-            Box::new(self.elements.first()?.0.get().get_type()?),
-            self.expanded_length(),
-        ))
+        let first = self.elements.first()?;
+        let inner_type = match first.0.get().get_type()? {
+            Type::Array(inner, _) if first.1 => *inner,
```

Comment:
whats this do? i don't think we want to flatten array types in asg?

Reply:
We would expect the following:
let a =[..."Hi", ..." world"]; // before compiled ok would be have a type of 2 [char; 2].
It would fix the following examples type inference:
let a = [...[1u8], 2u8]; // before compiled ok but said the type was an array of 2 [u8; 1].

Similarly this code would fail for type differences:
 let x = [1u8, 2u8];
let y = [...x, 3u8];
let z = y == [1u8; 3]; // before this would fail since lhs had type of 3 [u8; 2] and rhs had [u8; 3].